src/control/BarrettFingerKinematicSimulationPositionCommandExecutor.cpp

```diff
@@ -40,24 +40,35 @@ ::BarrettFingerKinematicSimulationPositionCommandExecutor(
   if (!mDistalDof)
     throw std::invalid_argument("Finger does not have distal dof.");
 
-  // Default mCollisionDetector to FCL collision detector
-  if (!mCollisionDetector)
-    mCollisionDetector = dart::collision::FCLCollisionDetector::create();
-
-  // Default mCollideWith to empty collision group. If a collision group is
-  // given and its collision detector does not match mCollisionDetector, set the
-  // collision group to an empty collision group.
-  if (!mCollideWith ||
-      mCollisionDetector != mCollideWith->getCollisionDetector())
+  if (mCollisionDetector && mCollideWith)
   {
-    if (mCollideWith)
+    // If a collision group is given and its collision detector does not match
+    // mCollisionDetector, set the collision group to an empty collision group.
```

Comment:
We could add `ShapeFrame`s of `collideWith` to the new collision group as
```cpp
mCollideWith = mCollisionDetector->createCollisionGroup();
for (auto i = 0u; i < collideWith->getNumShapeFrames(); ++i)
  mCollideWith->addShapeFrame(collideWith->getShapeFrame(i));
```
so that still can check collision as expected.